presto-tests/src/test/java/com/facebook/presto/server/TestQueryResource.java

```diff
@@ -351,4 +351,32 @@ private void assertStateCounts(List<BasicQueryInfo> infos, int expectedFinished,
         assertEquals(running, expectedRunning);
         assertEquals(queued, expectedQueued);
     }
+
+    @Test
+    public void testShouldReturnLegacyDataOnNestDataSerializationDisabled()
+            throws Exception
+    {
+        runner = createQueryRunner(ImmutableMap.of("nested-data-serialization-enabled", "false"));
+        server = runner.getCoordinator();
+
+        String sql = "SELECT MAP(ARRAY['a'], ARRAY[1])";
+        URI uri = uriBuilderFrom(server.getBaseUrl().resolve("/v1/statement")).build();
+
+        Request request = preparePost()
```

Comment:
I want to retrieve the data in its string format before it is deserialized. This will allow me to ensure that the returned string aligns with our expectations. I believe that the getQueryResults function uses the JSON codec, which will deserialize the response.

Reply:
Yeah, seems sensible to keep it as is since it's so specific to this test.